src/JSaddleDevice.ts

```diff
@@ -0,0 +1,153 @@
+import { BaseFileSystem, BFSCallback, FileSystem, FileSystemOptions, BFSOneArgCallback, BFSThreeArgCallback
+       } from "browserfs/dist/node/core/file_system";
+import Stats from 'browserfs/dist/node/core/node_fs_stats';
+import { BaseFile, File } from "browserfs/dist/node/core/file";
+import { FileFlag } from "browserfs/dist/node/core/file_flag";
+import { ApiError } from 'browserfs/dist/node/core/api_error';
+import { Device } from './DeviceFileSystem';
+
+export class JSaddleDevice implements Device {
+  private _file: JSaddleDeviceFile
+  constructor(
+    jsaddleListener: MessagePort,
+    jsaddleMsgBufArray: Uint8Array,
+    jsaddleMsgBufArray32: Uint32Array) {
+    this._file = new JSaddleDeviceFile(this, jsaddleListener
+                                  , jsaddleMsgBufArray, jsaddleMsgBufArray32);
```

Comment:
Why not just return the new file on `open`? Why cache it like this? Won't this have problems with closing and reopening?